src/js/actions/documents.js

```diff
@@ -1099,7 +1099,7 @@ define(function (require, exports) {
 
         return this.transfer(initActiveDocument);
     };
-    beforeStartup.reads = [locks.PS_DOC];
+    beforeStartup.reads = [locks.PS_DOC, locks.JS_APP];
```

Comment:
I don't see that this actually reads `JS_APP`. It gets a reference to the store up front, but that's it. Is this an important part of the fix?